citest/json_predicate/path_predicate.py

```diff
@@ -292,8 +315,8 @@ def __call__(self, context, source):
 
     path = context.eval(self.__path)
 
-    builder = PathPredicateResultBuilder(pred=self, source=source)
-    enumerate_terminal = True
+    builder = PathPredicateResultBuilder(pred=self.source_pred, source=source)
```

Comment:
#288 prevents source_pred from ever being None. It will default to self. Actually, if you explicitly set it to None then it will be None and result in an error. I can change #288 or be "or self" instead of defaulting to self.